### ISSUE TITLE
Some SysRPL apps or writed in assembler going to stuck

### ISSUE THREAD

**Issue by hpsaturn** [gwenhael-le-moine/x48ng#27]
## Summary

I understand that maybe are many pending task to migrate to this emulator. I only want to open this issue to try to understand what is missing for have a similar experience than other stable emulators like the Emu48 in Android. I notice that some special apps doesn't work well like Java library `STK` command, that the arrow keys (up/down)  in the menu are freezed but you can exit from the app:

![photo_2024-10-24_00-26-41](https://github.com/user-attachments/assets/fe8c20c8-624c-41d0-9d1f-efc5d327061a)
(I set these flags to hand using the SF/CF commands. With the STK option of Java was impossible)

**What we need to improve these kind of issues? I'm not expert in C but I can try to help. What is the road map or way to try to reach this?**

**Comment by gwenhael-le-moine** [comment]
I can use the up arrow to navigate but you're right that the down arrow doesn't seem to work.

I tested on the original x48 and it works there.
I also tested on the sister project [saturnng](https://github.com/gwenhael-le-moine/saturnng) where it also works (in `--sdl`, `--tui` is still buggy there, potentially for reasons linked to this bug.) (the UI is based on the one from x48ng but it has diverged a little, I plan to sync them back one day.)

I suspect the problem is not in the emulation itself but in how the UI handles keyboard events. This is where x48ng and saturnng mostly differ.

**Comment by hpsaturn** [comment]
yes maybe. Because in some games the keyboard events is weird and sometimes in some of those, doesn't work or is pretty buggy. In my case in the STK Java menu, I can't up or down.
